File: app/src/main/java/com/example/user/present/counter/count/UserPresentReceiver.java
package com.example.user.present.counter.count;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import androidx.annotation.NonNull;

import timber.log.Timber;

public class UserPresentReceiver extends BroadcastReceiver {

    private static final String TAG = "UserPresentReceiver";

    @Override
    public void onReceive(Context context, Intent intent) {
        Timber.d("onReceive: " + intent.getAction());

        Intent serviceIntent = new Intent(context.getApplicationContext(), RecordService.class);
        // フォアグラウンドサービスを起動中であればバックグラウンドサービスを起動することができる
        context.startService(serviceIntent);
    }

    @NonNull
    @Override
    public String toString() {
        return super.toString();
    }
}

File: app/src/main/java/com/example/user/present/counter/frameworks/home/UserPresentReceiver.kt
package com.example.user.present.counter.frameworks.home

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.example.user.present.counter.usecase.home.RecordSmartPhoneUsage
import timber.log.Timber

class UserPresentReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        Timber.d("onReceive: ${intent.action}")
        RecordSmartPhoneUsage(context.applicationContext).execute()
    }

    override fun toString(): String {
        return super.toString()
    }
}